applications/asset_tracker_v2/src/cloud/lwm2m_integration/lwm2m_integration.c

```diff
@@ -157,18 +156,22 @@ static void rd_client_event(struct lwm2m_ctx *client, enum lwm2m_rd_client_event
 		break;
 	case LWM2M_RD_CLIENT_EVENT_REG_UPDATE_FAILURE:
 		LOG_WRN("LWM2M_RD_CLIENT_EVENT_REG_UPDATE_FAILURE");
-		cloud_wrap_evt.type = CLOUD_WRAP_EVT_DISCONNECTED;
+		cloud_wrap_evt.type = CLOUD_WRAP_EVT_CONNECTING;
+		state = CONNECTING;
 		notify = true;
-		state = DISCONNECTED;
 		break;
 	case LWM2M_RD_CLIENT_EVENT_REG_UPDATE_COMPLETE:
 		LOG_DBG("LWM2M_RD_CLIENT_EVENT_REG_UPDATE_COMPLETE");
+		if (state == CONNECTING) {
```

Comment:
how about to keep disconnect state set add change this to state == CONNECTING || state == DISCONNECTED
It will be then smaller change. Or even remoceCONNECTING state and use DISCCONECTED at LWM2M_RD_CLIENT_EVENT_REG_UPDATE_FAILURE event also.

Reply:
I propose just if (state== DISCONNECTED)

Reply:
The most important part is that the application is able to recover if a failure happens.

Reply:
Discussed offline. Resolving.